src/query/arithmetic.c

```diff
@@ -5103,39 +5104,36 @@ db_evaluate_json_contains (DB_VALUE * result, DB_VALUE * const *arg, int const n
   if (path != NULL)
     {
       const char *raw_path = db_get_string (path);
-      JSON_DOC *extracted_doc = NULL;
 
+      JSON_DOC *extracted_doc = nullptr;
       /* *INDENT-OFF* */
-      error_code = db_json_extract_document_from_path (source_doc, std::vector<std::string> (1, raw_path), extracted_doc);
+      error_code = db_json_extract_document_from_path (source.get_borrowed (), {raw_path}, extracted_doc);
```

Comment:
Change db_json_extract_document_from_path signature (or overload) to output directly to JSON_DOC_WRAPPER.